### PR TITLE
Fixes #9618 Allow apostrophe in entity names

### LINKED ISSUE #9618: Allow apostrophe in entity names
It is currently not possible to use apostrophes in a glossary term's name (term is then invalid).
Lots of expressions in French use d' or l' and it is difficult not to use it in something that should reflect the business language.
Many thanks

### PR DESCRIPTION
**Pull request by sureshms**
### Describe your changes :

See #9618 for more details

#
### Type of change :

- [x] Bug fix
- [x] Backward Incompatible

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by sureshms** [comment]
@TeddyCr and @pmbrull, need help with reviewing this PR.

We were URL encoding KPI fqn, entityLink FQN in TestCase resource, and Ingestion Pipeline FQN. This is no longer required and now EntityLink allows spaces, and apostrophes in the names. Please review and let me know if this requires any upgrade changes.

@chirag-madlani need help with the Cypress test failures.

cc @harshach

**Comment by ShaileshParmar11** [comment]
Hi @sureshms We need a python side change 1st as all the cypress test is failing on test connection step!
cc @pmbrull

**Review by harshach — APPROVED** [review]

**Comment by ShaileshParmar11** [comment]
Custom property:
image

Glossary:

https://user-images.githubusercontent.com/71748675/211477071-bc13d46d-7381-41e2-9e8f-80bccf62ceae.mov

**Comment by TeddyCr** [comment]
@sureshms, if I well remember, the changes were put because entityLink and fqn are stored encoded in the DB, though the path parameters get decoded on the backend side which results in the request not returning any results as the fqn/entityLink filter does not match any values in the DB.

Let me know if that makes sense.

**Comment by sureshms** [comment]
> @sureshms, if I well remember, the changes were made because entityLink and fqn are stored encoded in the DB, though the path parameters get decoded on the backend side which results in the request not returning any results as the fqn/entityLink filter does not match any values in the DB.
> 
> Let me know if that makes sense.

@TeddyCr fqn are not stored as URL encoded string (especially where they are used in field_relationship tables). I have added comments in the patch with your name. Need help with this, please.

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/dqtests/TestCaseResource.java [hunk 1]
```diff
@@ -164,8 +162,7 @@ public ResultList<TestCase> list(
     ResourceContextInterface resourceContext;
     if (entityLink != null) {
       EntityLink entityLinkParsed = EntityLink.parse(entityLink);
-      filter.addQueryParam(
-          "entityFQN", URLEncoder.encode(entityLinkParsed.getFullyQualifiedFieldValue(), StandardCharsets.UTF_8));
+      filter.addQueryParam("entityFQN", entityLinkParsed.getFullyQualifiedFieldValue());
       resourceContext = TestCaseResourceContext.builder().entityLink(entityLinkParsed).build();
     } else {
```
Comment (sureshms) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/dqtests/TestCaseResource.java:167]: @TeddyCr, here filter param is being encoded. Given this is not stored in the DB and the tests are passing we should be good to go. Please confirm.

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/kpi/KpiResource.java [hunk 2]
```diff
@@ -387,8 +386,7 @@ public Response restoreKpi(
   public Response addKpiResult(
       @Context UriInfo uriInfo,
       @Context SecurityContext securityContext,
-      @Encoded @Parameter(description = "fqn of the kpi", schema = @Schema(type = "string")) @PathParam("fqn")
-          String fqn,
+      @Parameter(description = "fqn of the kpi", schema = @Schema(type = "string")) @PathParam("fqn") String fqn,
```
Comment (sureshms) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/kpi/KpiResource.java:389]: @TeddyCr, why should fqn be encoded here? It is used in findEntityByName(fqn) all the time for other entities where the fqn is not encoded. I also believe this does not have any upgrade impact. Please confirm.

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/services/ingestionpipelines/IngestionPipelineResource.java [hunk 3]
```diff
@@ -622,9 +621,7 @@ public Response getLastIngestionLogs(
   public Response addPipelineStatus(
       @Context UriInfo uriInfo,
       @Context SecurityContext securityContext,
-      @Encoded
-          @Parameter(description = "fqn of the ingestion pipeline", schema = @Schema(type = "string"))
```
Comment (sureshms) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/services/ingestionpipelines/IngestionPipelineResource.java:626]: @ulixius9 why is this parameter encoded? It should work without encoding. We need upgrade changes for this to rewrite pipeline entities fqn without url encoding, when we commit this change. cc @TeddyCr

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/services/ingestionpipelines/IngestionPipelineResource.java [hunk 4]
```diff
@@ -687,9 +684,7 @@ public ResultList<PipelineStatus> listPipelineStatuses(
   public PipelineStatus getPipelineStatus(
       @Context UriInfo uriInfo,
       @Context SecurityContext securityContext,
-      @Encoded
-          @Parameter(description = "fqn of the ingestion pipeline", schema = @Schema(type = "string"))
```
Comment (sureshms) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/services/ingestionpipelines/IngestionPipelineResource.java:691]: @ulixius9 again, URL encoding should not be necessary here.

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/dqtests/TestCaseResource.java (hunk 1 above)
Comment (TeddyCr) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/dqtests/TestCaseResource.java:167]: @sureshms that is my bad I should probably have added a new test, sorry about that. Basically the use case was for tables where column names have spaces in them. entityLink was previously saved as `"entityLink":"<#E::table::rds.dev.dbt_jaffle.table_col_w_space::columns::col%20space>"` (seems like in 0.13.1 this was removed as the entityLink is not encoded which raises an error @ShaileshParmar11). This results in the entityLink being saved encoded in the DB. When the GET request gets processed by the backend it get decoded and we have a miss match between entityLink in the db and entityLink requested.

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/kpi/KpiResource.java (hunk 2 above)
Comment (mohityadav766) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/kpi/KpiResource.java:389]: @sureshms yes , it need not be encoded for the KPI

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/dqtests/TestCaseResource.java (hunk 1 above)
Comment (TeddyCr) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/dqtests/TestCaseResource.java:167]: To add on to my previous comment, I tested the behavior with the latest [PR](https://github.com/open-metadata/OpenMetadata/pull/9668) against this file it seems to not break anything. We'll just need to adjust the supported pattern for the entityLink from `"^<#E::\\S+::\\S+>$"` to `"^<#E::\\S+::[\\S\\s]+>$"`.

https://github.com/open-metadata/OpenMetadata/blob/1ec324e43ed4eceb6a7505961d23d019c0cd5091/openmetadata-spec/src/main/resources/json/schema/type/basic.json#L88-L92

Screenshot 2023-01-11 at 2 52 10 PM

Comment (TeddyCr) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/dqtests/TestCaseResource.java:167]: Thanks @sureshms, makes sense. That should be good.

Comment (sureshms) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/dqtests/TestCaseResource.java:167]: 👍

**Comment by TeddyCr** [comment]
@sureshms -> commit for review with fixes https://github.com/open-metadata/OpenMetadata/pull/9642/commits/b1d0cb7e3f0e75f4d2e359b7a658c480d77ebed6

**Review by TeddyCr — APPROVED** [review]

**Comment by sureshms** [comment]
@TeddyCr, @pmbrull, it appears that we are using '/' in the entity names. Currently, this character clashes with URL path separator. Why is '/' required in the entity names? Is it for location?

**Review by harshach — APPROVED** [review]

**Pull request merged**